### PR TITLE
Tooltips for thumbnails in report pages

### PR DESCRIPTION
**Pull request by dohmatob**
Trying to make the report pages as comprehensive as possible to the front-end user.

### CODE REVIEW

**Comment by dohmatob** [comment]
screenshot from 2015-04-15 18 52 55
screenshot from 2015-04-15 18 44 16
screenshot from 2015-04-15 18 50 39

**Comment by dohmatob** [comment]
screenshot from 2015-04-15 20 49 28

**Comment by dohmatob** [comment]
Ping @alpinho: Is this OK for you ?

**Comment by bthirion** [comment]
The PR contains slightly disparate contributions. Altogether, this is an improvement.
OK for merging as soon as travis is happy.

File: pypreprocess/purepython_preproc_utils.py [hunk 1]
```diff
@@ -356,16 +300,14 @@ def do_subject_preproc(subject_data,
     func_basenames = [get_basenames(subject_data.func[sess]) if not is_niimg(
             subject_data.func[sess]) else "%s.nii.gz" % (
             subject_data.session_id[sess])
-                      for sess in xrange(subject_data.n_sessions)]
+                    for sess in xrange(subject_data.n_sessions)]
```
Comment (bthirion) [inline pypreprocess/purepython_preproc_utils.py]: xrange -> range

File: pypreprocess/reporting/glm_reporter.py [hunk 2]
```diff
@@ -338,7 +338,7 @@ def generate_subject_stats_report(
             design_thumbs.commit_thumbnails(thumb)
 
     # create activation thumbs
-    for contrast_id in contrasts:
+    for contrast_id, contrast_val in contrasts.iteritems():
```
Comment (bthirion) [inline pypreprocess/reporting/glm_reporter.py]: iteritems -> items

File: pypreprocess/slice_timing.py [hunk 3]
```diff
@@ -102,7 +102,7 @@ class STC(object):
         if set, then slices were acquired in interleaved order,
         odd-numbered slices first, and then even-numbered slices
 
-    ref_slice: int (optional, default 0)
+    refslice: int (optional, default 0)
```
Comment (bthirion) [inline pypreprocess/slice_timing.py]: ? I prefer ref_slice to refslice (or even better, reference_slice...)

File: pypreprocess/subject_data.py [hunk 4]
```diff
@@ -132,9 +148,8 @@ def _delete_orientation(self):
 
         # deleteorient for func
         self.func = [mem.cache(delete_orientation)(
-                self.func[sess],
-                self.session_output_dirs[sess])
-                     for sess in xrange(self.n_sessions)]
+            self.func[sess], self.session_output_dirs[sess])
+                    for sess in xrange(self.n_sessions)]
```
Comment (bthirion) [inline pypreprocess/subject_data.py]: xrange -> range

File: pypreprocess/subject_data.py [hunk 5]
```diff
@@ -189,15 +201,11 @@ def _niigz2nii(self):
...
         cache_dir = os.path.join(self.scratch, 'cache_dir')
         mem = Memory(cache_dir, verbose=100)
-
         self.func = [mem.cache(do_niigz2nii)(
-                self.func[sess],
-                output_dir=self.session_output_dirs[sess])
-                     for sess in xrange(self.n_sessions)]
-
+            self.func[sess], output_dir=self.session_output_dirs[sess])
+                    for sess in xrange(self.n_sessions)]
```
Comment (bthirion) [inline pypreprocess/subject_data.py]: idem

File: pypreprocess/subject_data.py [hunk 6]
```diff
@@ -326,22 +328,21 @@ def sanitize(self, deleteorient=False, niigz2nii=False):
...
             assert os.path.isfile(self.anat)
 
-        # sanitize session_id
-        if self.session_id is None:
+        # sanitize session_ids
+        if self.session_ids is None:
             if len(self.func) < 10:
-                self.session_id = ["session_%i" % i
+                self.session_ids = ["session_%i" % i
                                    for i in xrange(len(self.func))]
```
Comment (bthirion) [inline pypreprocess/subject_data.py]: idem

File: pypreprocess/subject_data.py [hunk 7]
```diff
@@ -360,15 +362,14 @@ def sanitize(self, deleteorient=False, niigz2nii=False):
 
         # get basenames
         self.basenames = [get_basenames(self.func[sess]) if not is_niimg(
-                self.func[sess]) else "%s.nii.gz" % self.session_id[sess]
-                          for sess in xrange(self.n_sessions)]
+                self.func[sess]) else "%s.nii.gz" % self.session_ids[sess]
+                        for sess in xrange(self.n_sessions)]
```
Comment (bthirion) [inline pypreprocess/subject_data.py]: idem

**Comment by bthirion** [comment]
Please address the details discussed here. 
Travis seems happy, so no further pb.

**Comment by dohmatob** [comment]
Done. Merging without further ado.

**Pull request merged**